### PR TITLE
Compare tests with oracle using the `diff` utility

### PR DESCRIPTION
**Pull request by benozol**
Preparation to remove the functionality from the colis executable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,6 +6,7 @@ build: extract-why3
 
 test: build
 	dune runtest
+	./run_tests.sh
```
Comment (Niols) [inline Makefile]: Actually, this should go in a Dune file rather than in the Makefile. But I'll take care of that eventually

File: run_tests.sh [hunk 2]
```diff
@@ -1,13 +1,69 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (Niols) [inline run_tests.sh]: Aw, sad :-(

Comment (Niols) [reply run_tests.sh]: Where do we need bash stuff?

File: run_tests.sh [hunk 3]
```diff
@@ -1,13 +1,69 @@
...
+    base=$(basename "$file" $extension)
+
+    # Derive oracle file name
+    oracle="tests/$base.oracle"
+    if ! [ -e "$oracle" ]; then
+        error "$file" "Missing oracle $oracle"
+        continue
     fi
+
+    outbase=$(mktemp -u -t "$base-XXXXX")
```
Comment (Niols) [inline run_tests.sh]: I've had problems before with `mktemp` not being available on all systems. But let's keep it that way for now, I'd say

File: Makefile (hunk 1 above)
Comment (benozol) [reply Makefile]: OK I’ll remove it from the PR

File: run_tests.sh (hunk 2 above)
Comment (benozol) [reply run_tests.sh]: Oh forgot to change it back, will do

File: run_tests.sh (hunk 3 above)
Comment (benozol) [reply run_tests.sh]: Is `tempfile` more generally available?

File: Makefile (hunk 1 above)
Comment (Niols) [reply Makefile]: I'm not sure you should. Maybe it should be the job of the future PR that'll pass tests under Dune?

**Comment by benozol** [comment]
And I was thinking to replace `RESULT: true|false` into `RETURN: <N>` in the oracle files. This would allow for oracle files that detect syntax errors or errors in the conversion. OK?

**Comment by Niols** [comment]
Yes, it's nice to have the integer. Where should we document which integer goes with which behaviour of the colis command?

**Comment by benozol** [comment]
What about describing the return codes in the README and/or the future man-page?

I left my computer and will probably not be able to review your other PR on the CLI before Monday :(

Bon weekend !

Am 5. Oktober 2018 16:06:14 MESZ schrieb Nicolas Jeannerod <notifications@github.com>:

**Pull request merged**